docs/devel/networking.md

```diff
@@ -1,3 +1,77 @@
+In this section we'll explain how VM networking is configured. In order to
+follow the principle of least privilege (where each component is limited
+to only its required privileges), the configuration of a KubeVirt VM interfaces
+is split into two distint phases:
+- [privileged networking configuration](#privileged-vmi-networking-configuration): occurs in the virt-handler process
+- [unprivileged networking configuration](#unprivileged-vmi-networking-configuration): occurs in the virt-launcher process
+
+### Privileged VMI networking configuration
+Virt-handler is a trusted component of KubeVirt; it runs on a privileged
+daemonset. It is responsible for creating & configuring any required network
+infrastructure (or configuration) required by the
+[binding mechanisms](#binding-mechanisms).
+
+It is important to refer that while this step is performed by virt-handler, it
+is performed in the *target* virt-launcher's net namespace, as can be seen in
+the implementation of the [setPodNetworkPhase1 function](https://github.com/kubevirt/kubevirt/blob/8e7f2a411f5d06f1734e96c7bc08d0bb9ec1d500/pkg/virt-handler/vm.go#L340).
+
+This will in turn call the
+[PlugPhase1](https://github.com/kubevirt/kubevirt/blob/5995d2164faefdcc7a59a857e12f8ed4cdb7e094/pkg/virt-launcher/virtwrap/network/podinterface.go#L84),
+which performs the following operations:
+- `getPhase1Binding`: this function initializes the interface structures, and
+  selects the correct `driver` to use. The driver, is essentially a
+  `BindMechanism`. The next operations will all be of specific implementations
+  of [BindMechanisms](#binding-mechanisms).
+
+- `discoverPodNetworkInterface`: Each `BindMechanism` requires different
+   information about the pod interface - slirp, for instance, doesn't require
+   any info. The others, gather the following information:
+   - IP address
+   - Routes (**only** bridge)
+   - Gateway
+   - MAC address (**only** bridge)
+   - Link MTU
+
+- `preparePodNetworkInterfaces`: this function will make use of the
+  aforementioned information, performing actions specific to each `BindMechanism`.
+  See each `BindMechanism` section for more details.
+
+- `setCachedInterface`: caches the interface in memory
+
+- `setCachedVIF`: this will persist the
+  [VIF](https://github.com/kubevirt/kubevirt/blob/51ce9d8e6def33c0c260f0cdb994f08ae5b9938e/pkg/virt-launcher/virtwrap/network/common.go#L48)
+  object in the file system, making the configured pod interface available to
+  the virt-launcher pod. The VIF is cached in
+  `/proc/<virt-launcher-pid>/root/var/run/kubevirt-private/vif-cache-<iface_name>.json`.
 
+### Unprivileged VMI networking configuration
+The virt-launcher is an untrusted component of KubeVirt (since it wraps the
+libvirt process that will run third party workloads). As a result, it must be
+run with as little privileges as required. As of now, the only capability
+required by virt-launcher to configure networking is the `CAP_NET_ADMIN`
```

Comment:
It actually also needs `CAP_NET_RAW` implicitly. (there is a PR/issue on that pending)

Reply:
I'm listing the network side capabilities we're currently using in the code - with currently being the keyword here.

Part of that pending PR should be to update this and add info about the `CAP_NET_RAW` capability.

Reply:
So maybe you can add this current fact: "CAP_NET_RAW is also a required capability but it is not explicitly mentioned at this time."
Not a must, just pointing out.